### PR TITLE
Add database API to list current executions

### PR DESCRIPTION
**Pull request by andrewjdawson2016**

### CODE REVIEW

File: common/persistence/persistenceMetricClients.go [hunk 1]
```diff
@@ -330,6 +330,20 @@ func (p *workflowExecutionPersistenceClient) GetCurrentExecution(request *GetCur
 	return response, err
 }
 
+func (p *workflowExecutionPersistenceClient) ListCurrentExecutions(request *ListCurrentExecutionsRequest) (*ListCurrentExecutionsResponse, error) {
+	p.metricClient.IncCounter(metrics.PersistenceListCurrentExecutionsScope, metrics.PersistenceRequests)
+
+	sw := p.metricClient.StartTimer(metrics.PersistenceGetCurrentExecutionScope, metrics.PersistenceLatency)
```
Comment (venkat1109) [inline common/persistence/persistenceMetricClients.go]: typo - fix Scope name

File: common/persistence/persistenceMetricClients.go [hunk 2]
```diff
@@ -330,6 +330,20 @@ func (p *workflowExecutionPersistenceClient) GetCurrentExecution(request *GetCur
...
+	response, err := p.persistence.ListCurrentExecutions(request)
+	sw.Stop()
+
+	if err != nil {
+		p.updateErrorMetric(metrics.PersistenceGetCurrentExecutionScope, err)
```
Comment (venkat1109) [inline common/persistence/persistenceMetricClients.go]: fix scope name

File: common/persistence/cassandra/cassandraPersistence.go [hunk 3]
```diff
@@ -437,6 +437,12 @@ workflow_state = ? ` +
 		`and visibility_ts = ? ` +
 		`and task_id = ?`
 
+	templateListCurrentExecutionsQuery = `SELECT domain_id, workflow_id, current_run_id ` +
+		`FROM executions ` +
+		`WHERE shard_id = ? ` +
+		`and type = ? ` +
+		`and run_id = ?`
```
Comment (yycptt) [inline common/persistence/cassandra/cassandraPersistence.go]: Tried this query on my local machine and got 
```
InvalidRequest: Error from server: code=2200 [Invalid query] message="PRIMARY KEY column "run_id" cannot be restricted as preceding column "domain_id" is not restricted"
```
Add a TODO here?

Comment (venkat1109) [reply common/persistence/cassandra/cassandraPersistence.go]: good catch. we actually need to filter on domainID - we have a cookie cutter domainID for currentExecutions

Comment (yycptt) [reply common/persistence/cassandra/cassandraPersistence.go]: Please correct me if I am wrong. The primary key for the table is `(shard_id, type, domain_id, workflow_id, run_id, visibility_ts, task_id)`. If we have filter on `run_id`, then I think all previous columns need to be filtered, which means we also need to add a filter on `workflow_id`.

**Comment by andrewjdawson2016** [comment]
Decided to implement this with visibility API instead of direct database scan.

**Pull request closed without merging**